test/sand/Sand.test.ts

```diff
@@ -1,38 +1,194 @@
+      const {userWithSand, usersWithoutSand} = await setupTest();
+      const userWithoutSand = usersWithoutSand[0];
+      const sandTransferValue = DECIMALS_18.mul(800);
+      await expect(
+        userWithSand.Sand.transfer(userWithoutSand.address, sandTransferValue)
+      ).to.be.revertedWith('not enough fund');
+    });
+
+    it('users should not be able to transfer token they dont have', async function () {
+      const {userWithSand, usersWithoutSand} = await setupTest();
+      const userWithoutSand = usersWithoutSand[0];
+      const sandTransferValue = DECIMALS_18.mul(1);
+      await expect(
+        userWithoutSand.Sand.transfer(userWithSand.address, sandTransferValue)
+      ).to.be.revertedWith('not enough fund');
+    });
+
+    it('users balance should not move if they transfer token to themselves', async function () {
+      const {Sand, userWithSand} = await setupTest();
+      const balanceUserWithsandBefore = await Sand.balanceOf(
+        userWithSand.address
+      );
+      const sandTransferValue = DECIMALS_18.mul(200);
+      userWithSand.Sand.transfer(userWithSand.address, sandTransferValue);
+      const balanceUserWithsandAfter = await Sand.balanceOf(
+        userWithSand.address
+      );
+      expect(balanceUserWithsandBefore).to.equal(balanceUserWithsandAfter);
+    });
+
+    it('total supply should not be affected by transfers', async function () {
+      const {
+        Sand,
+        sandBeneficiary,
+        userWithSand,
+        usersWithoutSand,
+      } = await setupTest();
+      const sandTransferValue = DECIMALS_18.mul(200);
+      sandBeneficiary.Sand.transfer(usersWithoutSand, sandTransferValue);
+      userWithSand.Sand.transfer(usersWithoutSand, sandTransferValue);
+      usersWithoutSand[0].Sand.transfer(usersWithoutSand, sandTransferValue);
+      const totalSupply = await Sand.totalSupply();
+      expect(totalSupply).to.equal(TOTAL_SUPPLY);
+    });
+  });
+
+  describe('Allowance', function () {
+    it('user should not be able to transfer more token than their allowance permit', async function () {
+      const {userWithSand, usersWithoutSand} = await setupTest();
+      userWithSand.Sand.approve(usersWithoutSand[0].address, 250);
+      await expect(
+        usersWithoutSand[0].Sand.transferFrom(
+          userWithSand.address,
+          usersWithoutSand[1].address,
+          350
+        )
+      ).to.be.revertedWith('Not enough funds allowed');
+    });
+
+    it('user should be able to transfer some tokens that their allowance permit', async function () {
+      const {userWithSand, usersWithoutSand} = await setupTest();
+      userWithSand.Sand.approve(usersWithoutSand[0].address, 250);
+      await expect(
+        usersWithoutSand[0].Sand.transferFrom(
+          userWithSand.address,
+          usersWithoutSand[1].address,
+          150
+        )
+      ).to.be.revertedWith('Not enough funds allowed');
```

Comment:
according to the test description this should not revert

Reply:
good catch thx!